### ISSUE TITLE
[CI] Deduplicate `actions/cache` directives

### ISSUE THREAD

**Issue by joshlf** [google/zerocopy#340]
We have a `generate_cache` job in CI which pre-populates a cache that is re-used by other jobs:

https://github.com/google/zerocopy/blob/3bb9a54ff458e104f7eb24beda7f8fb09794e1a5/.github/workflows/ci.yml#L410-L442

In each dependent job, we do the following:
- Use `needs: generate_cache`
- Use the following `actions/cache` action configuration:
  https://github.com/google/zerocopy/blob/3bb9a54ff458e104f7eb24beda7f8fb09794e1a5/.github/workflows/ci.yml#L77-L81

As of this writing, we copy that exact text five times. It'd be great if we could reduce the duplication.

## Mentoring instructions

As [proposed here](https://github.com/google/zerocopy/issues/340#issuecomment-1747242751) by @mgeisler:
- [ ] Create a separate `cache.yml` file in `.github/workflows`
- [ ] Populate it with a job that performs these steps:
  
  https://github.com/google/zerocopy/blob/3bb9a54ff458e104f7eb24beda7f8fb09794e1a5/.github/workflows/ci.yml#L77-L81
- [ ] Replace uses of that pattern in `ci.yml` with an invocation of that file, ie:
  ```yaml
  - name: Populate cache
    uses: `.github/workflows/cache.yml`
  ```

<details>
<summary>Old/obsolete mentoring instructions</summary>

As discussed in [this StackOverflow comment](https://stackoverflow.com/questions/77053736/give-alias-to-github-actions-uses-step?noredirect=1#comment135837054_77053736), 

> You can extract your own actions: [docs.github.com/en/actions/creating-actions](https://docs.github.com/en/actions/creating-actions). YAML's own support for this isn't available in GHA though: [github.com/actions/runner/issues/1182](https://github.com/actions/runner/issues/1182).

The task is to figure out whether there's a way to use that technique to define a new action in `ci.yml` that we can re-use in other steps. If there's an easy, simple syntax for this, we should do it. If it would require a lot of complexity, then it's not worth it.

</details>

**Comment by zoo868e** [comment]
Hi @joshlf,
Is `workflow_run` what you're looking for?

**Comment by zoo868e** [comment]
I apologize for any inconvenience. It appears that the `workflow_run` isn't triggered when the `Generate Cache Workflow` is initiated by a pull request. Is it acceptable to employ the following workaround for this situation?

https://github.com/orgs/community/discussions/25220#discussioncomment-3246922

**Comment by joshlf** [comment]
> I apologize for any inconvenience. It appears that the `workflow_run` isn't triggered when the `Generate Cache Workflow` is initiated by a pull request. Is it acceptable to employ the following workaround for this situation?
> 
> https://github.com/orgs/community/discussions/25220#discussioncomment-3246922

Thanks for finding that workaround! Honestly probably not worth the complexity given that this is a very minor problem to begin with. If there's no good solution, it's probably fine to just leave it as-is. I'm gonna close this, but feel free to comment if you figure something else out!

**Comment by mgeisler** [comment]
Would it perhaps be interesting to use a local action? I do that in Comprehensive Rust:

- [`.github/workflows/build.yml`](https://github.com/google/comprehensive-rust/blob/main/.github/workflows/build.yml#L43-L44) uses
- [`.github/workflows/setup-rust-cache/action.yml`](https://github.com/google/comprehensive-rust/blob/main/.github/workflows/setup-rust-cache/action.yml)

This has worked out great for us in the last 12 months or so.

**Comment by joshlf** [comment]
Oh yeah that's perfect! I'll re-open the issue with that suggestion. Thanks, @mgeisler!

**Comment by Sh0g0-1758** [comment]
@joshlf , I have created a PR as a patch for this issue. Can you please review it and suggest changes in it if required.

**Comment by dorryspears** [comment]
@joshlf Hello, made a fix for this with #835.  All tests pass
Happy weekend!
